wire-runtime/src/commonMain/kotlin/com/squareup/wire/ProtoReader.kt

```diff
@@ -90,13 +93,31 @@ class ProtoReader(private val source: BufferedSource) {
    * @param token value returned from the corresponding call to [beginMessage].
    */
   @Throws(IOException::class)
-  fun endMessage(token: Long) {
+  fun endMessageAndGetUnknownFields(token: Long): ByteString {
     check(state == STATE_TAG) { "Unexpected call to endMessage()" }
     check(--recursionDepth >= 0 && pushedLimit == -1L) { "No corresponding call to beginMessage()" }
     if (pos != limit && recursionDepth != 0) {
       throw IOException("Expected to end at $limit but was $pos")
     }
     limit = token
+    val unknownFieldsBuffer = bufferStack[recursionDepth]
+    return if (unknownFieldsBuffer.size > 0L) {
+      unknownFieldsBuffer.readByteString()
+    } else {
+      ByteString.EMPTY
+    }
+  }
+
+  /**
+   * End a length-delimited nested message. Calls to this method must be symmetric with calls to
+   * [beginMessage].
+   *
+   * @param token value returned from the corresponding call to [beginMessage].
+   */
+  @Throws(IOException::class)
+  @Deprecated(level = DeprecationLevel.HIDDEN, message = "prefer endMessageAndGetUnknownFields()")
```

Comment:
Wanna do WARNING deprecation? Code that has a compiler/runtime mismatch should still build?

Reply:
Good catch, code generated by previous versions of compiler will fail to compile after this change.